### ISSUE TITLE
Party members control

### ISSUE THREAD

**Issue by FakelsHub** [phobos2077/sfall#179]
It is required to refine the mod of party member control.

1. Make the centering of the screen on the selected party member.
2. To remove the tile roofs when give control to party member.
3. Add a pointing label with the name of the party member.

**Comment by FakelsHub** [comment]
2. tile_refresh_display ???
3. yeah, I didn't think of that.

**Comment by NovaRain** [comment]
2. I tried it, but it doesn't seem to work.

**Comment by FakelsHub** [comment]
3. -1 disabled.

**Comment by FakelsHub** [comment]
There are problems in the script, the move_to function is executed twice, this causes a flicker of the picture, at the beginning of the round of the combat there is an unnecessary centering on the player, and show a name box.
Here is the modified code for the script, just paste in place of the old one.
```
variable npcIsControl;
//-----------
   if (status == 1
      and (len_array(pidList) == 0 or scan_array(pidList, pid bwand 0xFFFFFF) != -1)
      and (controlMode == 1 or party_member_obj(pid))) then begin
      if not(critter == real_dude_obj) then begin
         set_dude_obj(critter);
         if hasGeckoSkinning then critter_add_trait(critter, TRAIT_PERK, PERK_gecko_skinning_perk, 1);
         npcIsControl  := true;
      end
      if (npcIsControl) then begin
          // center the screen on the controlled critter and remove roof tiles
          move_to(dude_obj, dude_tile, dude_elevation);
          if (displayName) then begin
              set_iface_tag_text(displayName, obj_name(critter), displayNameColor);
              show_iface_tag(displayName);
          end
      end
   end else if (npcIsControl) then begin
      set_dude_obj(real_dude_obj);
      hide_iface_tag(displayName);
      if (status == -1) then npcIsControl  := false;
   end
end
```

**Comment by FakelsHub** [comment]
Maybe need gives to npc the Awareness perk, as it's done with Gecko Skinning

**Comment by NovaRain** [comment]
>Here is the modified code for the script, just paste in place of the old one.

With NPC control mode 2, your code makes the player not be able to control PC in combat after the first turn, and PC will be taken by another NPC. I'll just put displayName and move_to inside ```if not(critter == real_dude_obj)``` block.

**Comment by NovaRain** [comment]
>Here is the modified code for the script, just paste in place of the old one.

Is npcIsControl outside of combatturn_handler or inside? If inside, the player won't be able to control PC after the first turn, and PC will be taken by other NPC. If outside and the previous combat ends normally (ended by the game itself), at the start of the next combat the screen will center on PC with name box.

**Comment by FakelsHub** [comment]
Need to reset the variable through another hook - GameModeChange.

**Comment by NovaRain** [comment]
>UDT: if 3 argument changes? then it can be used.

Nah, it doesn't change. I'll just move ```move_to``` line into ```if (status == 1``` block to prevent the screen flickering and put displayName in ```if not(critter == real_dude_obj)``` block to not show name box while controlling PC. Yes, the screen still centers on PC at the very beginning of a player-initiated combat, but at least it's much less annoying than the previous version.

**Comment by NovaRain** [comment]
>Need to reset the variable through another hook - GameModeChange.

Oh, ok. I didn't think of that.
```
   end else then begin
      set_dude_obj(real_dude_obj);
      if (displayName) then hide_iface_tag(displayName);
   end
end

procedure gamemodechange_handler begin
   if not(get_game_mode BWAND COMBAT) then npcIsControl := false;
end
```

**Comment by FakelsHub** [comment]
need check a combat, or it will be reset when changing the mode.

**Comment by FakelsHub** [comment]
Well everything is fine.
